app/jobs/inherit_permissions_job.rb

```diff
@@ -4,21 +4,64 @@ class InheritPermissionsJob < Hyrax::ApplicationJob
   # Perform the copy from the work to the contained filesets
   #
   # @param work containing access level and filesets
-  def perform(work)
-    work.file_sets.each do |file|
-      attribute_map = work.permissions.map(&:to_hash)
+  # @param use_valkyrie [Boolean] whether to use valkyrie support
+  def perform(work, use_valkyrie: Hyrax.config.use_valkyrie?)
+    if use_valkyrie
+      work_permissions = permissions_map(work.permission_manager.acl.permissions)
```

Comment:
I like that you extracted the `af_perform` method. Would you be up for extracting a private method concerning `valkyrie_perform`?

Reply:
:+1:  Done.